### ISSUE TITLE
Rotating Element Mount / Gimbal

### ISSUE THREAD

**Issue by samdeane** [samedicorp/dualuniverse-feedback#32]
An element that could have (an)other element(s) attached to it and which could rotate.

This would allow for ship designs with rotating engines, wings, turrets, etc, as well as fancy building designs.
 
Rather than rotating engines per-se, could we have some sort of rotating mount that you anchor any other element to, to allow it to rotate?
 
Should be scriptable too, obvs :)

**Comment by samdeane** [comment]
https://upvote.dualuniverse.game/suggestions/125420/rotating-element-mount-gimbal

**Comment by Wolw3rin3** [comment]
Yeah I was thinking about that, was not sure is it to much to ask.

**Comment by samdeane** [comment]
It could be difficult to implement, for sure. I think eventually they will want to do it somehow, because there are so many nice ship behaviours it would unlock. 

Maybe the way they do it instead is to have a separate element for any moveable part - eg “Moveable Atmospheric Engine S”, “Moveable Wing S”, etc. In some ways that’s more work for the coders and designers, and it’s definitely more work for us poor users who have to set up factories to build them all :). However, it’s a more low-tech solution and it would allow them to hand-tune the behaviour in each case.
